plugins/samples/set_cookie/plugin.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_set_cookie]
+use proxy_wasm::traits::*;
+use proxy_wasm::types::*;
+use rand::rngs::SmallRng;
+use rand::{RngCore, SeedableRng};
+
+// Define the cookie name as a constant
+const COOKIE_NAME: &str = "my_cookie";
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Info);
+    proxy_wasm::set_root_context(|_| -> Box<dyn RootContext> {
+        let seed = std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap_or_default()
+            .as_nanos() as u64;
+
+        Box::new(MyRootContext {
+            rng: SmallRng::seed_from_u64(seed),
```

Comment:
I think you need to make this a std::rc::Rc<SmallRng> and use the same shared SmallRng instance from MyHttpContext. Otherwise, this is repeatedly clone'ing SmallRng with a given seed, so each MyHttpContext will see the same "random" value from its calls to next_u32().